crates/task-impls/src/quorum_vote/handlers.rs

```diff
@@ -205,11 +202,15 @@ pub(crate) async fn handle_quorum_proposal_validated<
                             "Failed to convert the serialized QC signature into a DRB seed input."
                         );
                     };
-                    let new_drb_task =
-                        spawn(async move { compute_drb_result::<TYPES>(drb_seed_input) });
+
+                    // This will join_or_abort_task before starting the new task
                     task_state
                         .drb_computations
-                        .insert(new_epoch_number, new_drb_task);
+                        .start_new_task(new_epoch_number, drb_seed_input)
+                        .await;
+                } else {
+                    // Only needed if we aren't starting a new task
+                    task_state.drb_computations.join_or_abort_task().await;
```

Comment:
The `else` case is where we aren't on the committee of the next epoch, so we don't have to handle it explicitly.

Is this the only place where `join_or_abort_task` is used? If so we could make it a private function.